ipatests/test_integration/test_commands.py

```diff
@@ -523,6 +523,28 @@ def test_huge_password(self):
                                     raiseonerr=False)
         assert result.returncode == 0
 
+    def test_cleartext_password_httpd_log(self):
+        """Test to check password leak in apache error log
+
+        Host enrollment with OTP used to log the password in cleartext
+        to apache error log. This test ensures that the password should
+        not be log in cleartext.
+
+        related: https://pagure.io/freeipa/issue/8017
+        """
+        hostname = 'test.{}'.format(self.master.domain.name)
+        passwd = 'Secret123'
+
+        self.master.run_command(['ipa', 'host-add', '--force',
+                                 hostname, '--password', passwd])
+
+        result = self.master.run_command(['grep', 'host_add',
+                                          paths.VAR_LOG_HTTPD_ERROR])
+        assert (hostname in result.stdout_text and
+                passwd not in result.stdout_text)
+
+        self.master.run_command(['ipa', 'host-del', hostname])
```

Comment:
Please move the host-del before the assert so that things will always be cleaned up, even in case the assert fails.

The grep is rather broad and may return multiple, unrelated entries, but in any case the password should never be logged so this should be fine.

Reply:
Changed the grep to grab the entries containing host only. This should minimize the result.